server/core/region.go

```diff
@@ -437,165 +437,47 @@ func (r *RegionInfo) GetReplicationStatus() *replication_modepb.RegionReplicatio
-	if old, ok := rm.m[region.GetID()]; ok {
-		rm.totalSize -= old.approximateSize
-		rm.totalKeys -= old.approximateKeys
-	}
-	rm.m[region.GetID()] = region
-	rm.totalSize += region.approximateSize
-	rm.totalKeys += region.approximateKeys
+func (rm regionMap) Put(region *RegionInfo) {
+	rm[region.GetID()] = region
 }
 
-func (rm *regionMap) Delete(id uint64) {
-	if rm == nil {
-		return
-	}
-	if old, ok := rm.m[id]; ok {
-		delete(rm.m, id)
-		rm.totalSize -= old.approximateSize
-		rm.totalKeys -= old.approximateKeys
-	}
-}
-
-func (rm *regionMap) TotalSize() int64 {
-	if rm.Len() == 0 {
-		return 0
-	}
-	return rm.totalSize
-}
-
-// regionSubTree is used to manager different types of regions.
-type regionSubTree struct {
-	*regionTree
-	totalSize int64
-	totalKeys int64
-}
-
-func newRegionSubTree() *regionSubTree {
-	return &regionSubTree{
-		regionTree: newRegionTree(),
-		totalSize:  0,
-	}
-}
-
-func (rst *regionSubTree) TotalSize() int64 {
-	if rst.length() == 0 {
-		return 0
-	}
-	return rst.totalSize
-}
-
-func (rst *regionSubTree) scanRanges() []*RegionInfo {
-	if rst.length() == 0 {
-		return nil
-	}
-	var res []*RegionInfo
-	rst.scanRange([]byte(""), func(region *RegionInfo) bool {
-		res = append(res, region)
-		return true
-	})
-	return res
-}
-
-func (rst *regionSubTree) update(region *RegionInfo) {
-	overlaps := rst.regionTree.update(region)
-	rst.totalSize += region.approximateSize
-	rst.totalKeys += region.approximateKeys
-	for _, r := range overlaps {
-		rst.totalSize -= r.approximateSize
-		rst.totalKeys -= r.approximateKeys
-	}
-}
-
-func (rst *regionSubTree) remove(region *RegionInfo) {
-	if rst.length() == 0 {
-		return
-	}
-	if rst.regionTree.remove(region) != nil {
-		rst.totalSize -= region.approximateSize
-		rst.totalKeys -= region.approximateKeys
-	}
-}
-
-func (rst *regionSubTree) length() int {
-	if rst == nil {
-		return 0
-	}
-	return rst.regionTree.length()
-}
-
-func (rst *regionSubTree) RandomRegion(ranges []KeyRange) *RegionInfo {
-	if rst.length() == 0 {
-		return nil
-	}
-
-	return rst.regionTree.RandomRegion(ranges)
-}
-
-func (rst *regionSubTree) RandomRegions(n int, ranges []KeyRange) []*RegionInfo {
-	if rst.length() == 0 {
-		return nil
-	}
-
-	regions := make([]*RegionInfo, 0, n)
-
-	for i := 0; i < n; i++ {
-		if region := rst.regionTree.RandomRegion(ranges); region != nil {
-			regions = append(regions, region)
-		}
-	}
-	return regions
+func (rm regionMap) Delete(id uint64) {
+	delete(rm, id)
 }
 
 // RegionsInfo for export
 type RegionsInfo struct {
 	tree         *regionTree
-	regions      *regionMap                // regionID -> regionInfo
-	leaders      map[uint64]*regionSubTree // storeID -> regionSubTree
-	followers    map[uint64]*regionSubTree // storeID -> regionSubTree
-	learners     map[uint64]*regionSubTree // storeID -> regionSubTree
-	pendingPeers map[uint64]*regionSubTree // storeID -> regionSubTree
+	regions      regionMap              // regionID -> regionInfo
```

Comment:
can use map directly

Reply:
This is a split PR, and there are other changes to the function of the subsequent regionMap, which will become complicated.

Reply:
cool.